src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that stores user recommendation data in datastore */
+@WebServlet("/recommendation")
+public class RecommendationServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    String name = request.getParameter("place-name");
+    String formattedLatLng = request.getParameter("location");
+
+    // Get latitude and longitude from formattedLatLng
+    String [] latLng = formattedLatLng.split(", ");
+    double lat = Double.parseDouble(latLng[0]);
+    double lng = Double.parseDouble(latLng[1]);
+    String category = request.getParameter("category-list");
```

Comment:
Nit: Maybe you don't have time for this now but later please consider adding a validator method to check if all the parameters are valid. For example, you are taking a string here, which can be _anything_, as a category but I assume you expect a definite list of strings a category.